core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/named/cache/NamedOperationCache.java

```diff
@@ -145,36 +138,23 @@ public NamedOperationDetail getFromCache(final String name) throws CacheOperatio
      *                                 or if the add operation fails for some reason.
      */
     @SuppressFBWarnings(value = "DCN_NULLPOINTER_EXCEPTION", justification = "Investigate an improved null checking approach")
-    public void addNamedOperation(final NamedOperationDetail namedOperation, final boolean overwrite, final User user,
-                                  final String adminAuth)
-            throws CacheOperationException {
-        String name;
-        try {
-            name = namedOperation.getOperationName();
-        } catch (final NullPointerException e) {
-            throw new CacheOperationException("NamedOperation cannot be null", e);
-        }
-        if (Objects.isNull(name)) {
-            throw new CacheOperationException("NamedOperation name cannot be null");
+    public void addNamedOperation(final NamedOperationDetail namedOperation, final boolean overwrite, final User user, final String adminAuth) throws CacheOperationException {
+        if (namedOperation == null) {
+            throw new CacheOperationException("NamedOperation cannot be null");
         }
-        if (!overwrite) {
-            addToCache(name, namedOperation, false);
-            return;
+        String name = namedOperation.getOperationName();
+        if (contains(name) && overwrite) {
+            final boolean doesUserHavePermissionToWrite = getFromCache(name).hasWriteAccess(user, adminAuth);
```

Comment:
Shouldn't user need write access to perform normal add also (not overwriting)?

Reply:
So in this situation.

I add a NamedOperation to cache, but I specify you have write permissions. This means you can override my NamedOperation. 

Else where in Gaffer land there will be checks for if you can use the AddNamedOperation, But here is where a 3rd person is stopped from overriding my namedOperation, but who has AddNamedOperation permission.

Reply:
Is it possible to add a NamedOperation that I will not be able to access later?

Reply:
This shouldn't happen and if it does it is an unrelated preexisting issue